fdbserver/workloads/Mako.actor.cpp

```diff
@@ -0,0 +1,530 @@
+			opCounters.push_back(PerfIntCounter(opNames[i]));
+		}
+	}
+
+	std::string description() override {
+		// Mako is a simple workload to measure the performance of FDB.
+		// The primary purpose of this benchmark is to generate consistent performance results
+		return "Mako";
+	}
+
+	Future<Void> setup(Database const& cx) override {
+		// populate data when we are in "build" mode and use one client to do it
+		if (mode.compare("build") == 0 && clientId == 0)
+			return _setup(cx, this);
+		return Void();
+	}
+
+	Future<Void> start(Database const& cx) override {
+		// TODO: Do I need to read data to warm the cache of the keySystem like ReadWrite.actor.cpp (line 465)?
+		if (mode == "clean"){
+			if (clientId == 0)
+				return cleanup(cx, this);
+			else
+				return Void();
+		}
+
+		return _start(cx, this);
+	}
+
+	Future<bool> check(Database const& cx) override {
+		return true;
+	}
+
+	void getMetrics(vector<PerfMetric>& m) override {
+		if (mode == "clean"){
+			return;
+		}
+		// Add metrics of population process
+		if (mode == "build" && clientId == 0){
+			m.push_back( PerfMetric( "Mean load time (seconds)", loadTime, true ) );
+			// The importing rate of keys, controlled by parameter "insertionCountsToMeasure"
+			auto ratesItr = ratesAtKeyCounts.begin();
+			for(; ratesItr != ratesAtKeyCounts.end(); ratesItr++){
+				m.push_back(PerfMetric(format("%ld keys imported bytes/sec", ratesItr->first), ratesItr->second, false));
+			}
+		}
+		m.push_back(PerfMetric("Measured Duration", metricsDuration, true));
+		m.push_back(xacts.getMetric());
+		m.push_back(PerfMetric("Transactions/sec", xacts.getValue() / testDuration, true));
+		m.push_back(totalOps.getMetric());
+		m.push_back(PerfMetric("Operations/sec", totalOps.getValue() / testDuration, true));
+		m.push_back(conflicts.getMetric());
+		m.push_back(PerfMetric("Conflicts/sec", conflicts.getValue() / testDuration, true));
+		m.push_back(retries.getMetric());
+
+		// count of each operation
+		for (int i = 0; i < MAX_OP; ++i){
+			m.push_back(opCounters[i].getMetric());
+		}
+
+		// Meaningful Latency metrics
+		const int opExecutedAtOnce[] = {OP_GETREADVERSION, OP_GET, OP_GETRANGE, OP_SGET, OP_SGETRANGE, OP_COMMIT};
+		for (const int& op : opExecutedAtOnce){
+			m.push_back(PerfMetric("Mean " + std::string(opNames[op]) +" Latency (ms)", 1000 * opLatencies[op].mean(), true));
+			m.push_back(PerfMetric("Max " + std::string(opNames[op]) + " Latency (ms, averaged)", 1000 * opLatencies[op].max(), true));
+			m.push_back(PerfMetric("Min " + std::string(opNames[op]) + " Latency (ms, averaged)", 1000 * opLatencies[op].min(), true));
+		}
+
+		//insert logging metrics if exists
+		m.insert(m.end(), periodicMetrics.begin(), periodicMetrics.end());
+	}
+	static void randStr(std::string& str, const int& len) {
+		static char randomStr[BUFFERSIZE];
+		for (int i = 0; i < len; ++i) {
+			randomStr[i] = '!' + g_random->randomInt(0, 'z' - '!'); /* generage a char from '!' to 'z' */
```

Comment:
Just want to expose the code and one of our team member at snowflake will review it. So I just mark it as a draft. Sorry for the confusion.

Reply:
We may do a profiling later for it. 
Currently, the two metrics in this benchmark are `TPS` and `Latency`. 
For `TPS`, which depends on the server-side,  even this one adds more delay for client-side, we can use more testers to get the right `TPS` .
For `Latency`, it may add delay to the result. But our goal here is the consistency of the test, we care more about the difference between releases or configs. Even the absolute values of `Latency` may lose some accuracy, it doesn't matter for our comparison.